### ISSUE TITLE
Refactor `plot_bandstructure` interface

### ISSUE THREAD

**Issue by mfherbst** [JuliaMolSim/DFTK.jl#503]
I've become a little unhappy about the `plot_bandstructure` interface, especially the fact that the kline_density has so strange units.
When Brillouin.jl is used this should be refactored.

**Comment by thchr** [comment]
Yeah, I had just briefly seen it the other day: cool effort, definitely!

One thing that came to mind when I was browsing the code of AtomsBase.jl was that it might be nice for various crystallography-related packages to settle on some standard "back-bone" for how to specify and transform direct and reciprocal lattices. The motivation would be to reduce the setting-related uncertainty that we also discussed in #496.
The implementations in Bravais might be suitable in that regard (e.g., in place of `Box`).

**Comment by mfherbst** [comment]
> The implementations in Bravais might be suitable in that regard (e.g., in place of Box).

Feel free to raise that aspect in `AtomsBase.jl`. Since AtomsBase is meant to be a bit more general (i.e. including molecular systems) I suppose it Bravais might not yet fit at that level, but I could imagine it to make sense to have a slightly more specialised implementation just for periodic problems ... and at that level Bravais could fit very well, I agree.

**Comment by mfherbst** [comment]
See the code in https://github.com/thchr/Brillouin.jl/blob/master/src/requires/plotlyjs_dispersion.jl
